ee/api/test/test_team_memberships.py

```diff
@@ -0,0 +1,380 @@
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+        self.assertEqual(self.team.explicit_memberships.count(), 0)
+
+    def test_add_yourself_as_org_admin_forbidden(self):
+        self.organization_membership.level = OrganizationMembership.Level.ADMIN
+        self.organization_membership.save()
+
+        self.assertEqual(self.team.explicit_memberships.count(), 0)
+
+        response = self.client.post("/api/projects/@current/explicit_members/", {"user_uuid": self.user.uuid})
+        response_data = response.json()
+
+        self.assertDictEqual(
+            self.permission_denied_response("You can't explicitly add yourself to projects."), response_data
+        )
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+        self.assertEqual(self.team.explicit_memberships.count(), 0)
+
+    def test_add_member_as_org_member_and_project_member_forbidden(self):
+        self.organization_membership.level = OrganizationMembership.Level.MEMBER
+        self.organization_membership.save()
+        ExplicitTeamMembership.objects.create(
+            team=self.team, parent_membership=self.organization_membership, level=ExplicitTeamMembership.Level.MEMBER
+        )
+
+        new_user: User = User.objects.create_and_join(self.organization, "rookie@posthog.com", None)
+
+        self.assertEqual(self.team.explicit_memberships.count(), 1)
+
+        response = self.client.post("/api/projects/@current/explicit_members/", {"user_uuid": new_user.uuid})
+        response_data = response.json()
+
+        self.assertDictEqual(
+            self.permission_denied_response("You don't have sufficient permissions in this project."), response_data
+        )
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+        self.assertEqual(self.team.explicit_memberships.count(), 1)
+
+    def test_add_member_as_org_member_but_project_admin_allowed(self):
+        self.organization_membership.level = OrganizationMembership.Level.MEMBER
+        self.organization_membership.save()
+        ExplicitTeamMembership.objects.create(
+            team=self.team, parent_membership=self.organization_membership, level=ExplicitTeamMembership.Level.ADMIN
+        )
+
+        self.assertEqual(self.team.explicit_memberships.count(), 1)
+
+        new_user: User = User.objects.create_and_join(self.organization, "rookie@posthog.com", None)
+
+        response = self.client.post("/api/projects/@current/explicit_members/", {"user_uuid": new_user.uuid})
+        response_data = response.json()
+
+        self.assertDictContainsSubset(
+            {"effective_level": ExplicitTeamMembership.Level.MEMBER, "level": ExplicitTeamMembership.Level.MEMBER,},
+            response_data,
+        )
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        self.assertEqual(self.team.explicit_memberships.count(), 2)
+
+    def test_add_member_as_org_admin_and_project_member_allowed(self):
```

Comment:
Looking at this test case, should we maybe restrict that admins and owners can't be added as project members? This way we avoid weird edge cases for when the user's org permissions changes. It also avoids extra handling such as `test_add_admin_as_project_member_forbidden`

Reply:
IMO it doesn't make sense to normalize this further, as then we easily end up in a situation where a previous plain organization member gets added to a project and elevated org-wide later on, in which case checks project-member-addition-time are moot.

Reply:
My concern is from a UX-standpoint, it's confusing that if you're an admin/owner you can be added to projects only under certain conditions. It's simpler to communicate that you can never be added to projects if you're an admin because you have access to everything.

Reply:
I totally agree UX-wise! The UI won't allow adding any admin/owner to projects. But in the backend I worry the code would just get more complex kind of for no significant gain.
BTW `test_add_admin_as_project_member_forbidden` definitely can't be avoided – the name definitely is a bit ambiguous as I had to keep it pretty concise, but what it tests is that, being a plain project member, you can't add another organization member (who doesn't belong to the project yet) as a project admin.

Reply:
Got it! Well if we don't allow it in the frontend, I'm totally fine with this!